srcts/src/components/darkModeSwitch.ts

```diff
@@ -0,0 +1,323 @@
+import type { CSSResultGroup } from "lit";
+import { LitElement } from "lit";
+import { css, html } from "lit";
+import { property } from "lit/decorators.js";
+
+// Inspired by:
+// https://web.dev/building-a-theme-switch-component/
+// https://web.dev/patterns/theming/theme-switch/
+// https://github.com/argyleink/gui-challenges/tree/main/theme-switch
+
+export type DarkModeMessageToggle = {
+  method: "toggle";
+  value: DarkModeSwitch["mode"] | undefined;
+};
+
+export class DarkModeSwitch extends LitElement {
```

Comment:
The standard naming convention for custom elements is for the class name to be the same as the element name, except the class is PascalCase, and the element is kebab-case.


```suggestion
export class BslibInputDarkMode extends LitElement {
```